src/ripple/protocol/Serializer.h

```diff
@@ -280,6 +282,20 @@ class Serializer
     int addEncoded (int length);
 };
 
+template<class FwdIt>
+int Serializer::addVL(FwdIt begin, FwdIt end, int len)
+{
+    int ret = addEncoded(len);
+    if (len)
+        for (; begin != end; ++begin)
```

Comment:
Because `len` is wrong: imagine an array `uint128[16]`. Then `len` is passed in as `16 * 128` but `std::distance` on the iterators will give you `16`.